src/types/Claims.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import { SplitComponent } from "./Components.sol";
+
+struct BasicTransfer {
+    bytes allocatorSignature; // Authorization from the allocator.
+    uint256 nonce; // A parameter to enforce replay protection, scoped to allocator.
+    uint256 expires; // The time at which the transfer or withdrawal expires.
+    uint256 id; // The token ID of the ERC6909 token to transfer or withdraw.
+    uint256 amount; // The token amount to transfer or withdraw.
+    address recipient; // The recipient of the transfer or withdrawal.
+}
+
+struct SplitTransfer {
+    bytes allocatorSignature; // Authorization from the allocator.
+    uint256 nonce; // A parameter to enforce replay protection, scoped to allocator.
+    uint256 expires; // The time at which the transfer or withdrawal expires.
+    uint256 id; // The token ID of the ERC6909 token to transfer or withdraw.
+    SplitComponent[] recipients; // The recipients and amounts of each transfer.
+}
+
+struct BasicClaim {
+    bytes allocatorSignature; // Authorization from the allocator.
+    bytes sponsorSignature; // Authorization from the sponsor.
```

Comment:
they're at the beginning so that the pointers are always present at specific calldata offsets, regardless of the struct; agree that the ordering is a bit weird, but it's also never directly exposed to end users

there are definitely cases where you can only provide 65 byte signatures, plus we want to support 1271 schemes that might not even pass in an ECDSA signature or need extra data